### PR TITLE
JAMES-2166 Generate Message-Id header while creating a message via JMAP

### PR DESCRIPTION
**Pull request by rouazana**

### CODE REVIEW

File: server/protocols/jmap/src/test/java/org/apache/james/jmap/methods/MIMEMessageConverterTest.java [hunk 1]
```diff
@@ -406,7 +466,7 @@ public void convertShouldGenerateExpectedMultipartWhenHtmlAndTextBodyProvided()
 
         String expectedHeaders = "MIME-Version: 1.0\r\n" +
                 "Content-Type: multipart/alternative;\r\n" +
-                " boundary=\"-=Part.1.";
+                " boundary=\"-=Part.";
```
Comment (mbaechler) [inline server/protocols/jmap/src/test/java/org/apache/james/jmap/methods/MIMEMessageConverterTest.java]: is it related ?

Comment (rouazana) [reply server/protocols/jmap/src/test/java/org/apache/james/jmap/methods/MIMEMessageConverterTest.java]: yes more or less, both are using MimeUtil which is using (between others) a static counter to get a unique id. The same counter is used for generating boundaries and message ids. Of course that's not the only thing that makes the id unique.

Comment (mbaechler) [reply server/protocols/jmap/src/test/java/org/apache/james/jmap/methods/MIMEMessageConverterTest.java]: Sorry to be picky, but that means the assertions is too strong and we should weaken it

Comment (rouazana) [reply server/protocols/jmap/src/test/java/org/apache/james/jmap/methods/MIMEMessageConverterTest.java]: that's exactly what I did

**Comment by chibenwa** [comment]
:+1:

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**